File: src/main/java/xyz/devvydont/smprpg/entity/vanilla/LeveledRavager.java
package xyz.devvydont.smprpg.entity.vanilla;

import org.bukkit.entity.Ravager;
import xyz.devvydont.smprpg.entity.MobType;
import xyz.devvydont.smprpg.entity.base.VanillaEntity;

public class LeveledRavager extends VanillaEntity<Ravager> {

    public LeveledRavager(Ravager entity) {
        super(entity);
    }

    @Override
    public void setup() {
        mobTypes.add(MobType.RARE);
        mobTypes.add(MobType.ILLAGER);
        mobTypes.add(MobType.ANIMAL);

        super.setup();
    }
}

File: src/main/java/xyz/devvydont/smprpg/entity/vanilla/LeveledRavager.kt
package xyz.devvydont.smprpg.entity.vanilla

import org.bukkit.entity.Ravager
import org.bukkit.persistence.PersistentDataType
import xyz.devvydont.smprpg.entity.MobType
import xyz.devvydont.smprpg.entity.base.VanillaEntity
import xyz.devvydont.smprpg.entity.slayer.illager.IllagerWarlockParent
import xyz.devvydont.smprpg.util.persistence.KeyStore

class LeveledRavager(entity: Ravager?) : VanillaEntity<Ravager?>(entity) {
    override fun setup() {
        mobTypes.add(MobType.RARE)
        mobTypes.add(MobType.ILLAGER)
        mobTypes.add(MobType.ANIMAL)

        super.setup()

        _entity!!.persistentDataContainer.set(
            KeyStore.SLAYER_SPAWN_TYPE,
            PersistentDataType.STRING,
            IllagerWarlockParent.SPAWN_MOB_FLAG
        )
    }
}
